url21b471891f_Agetac_Model_tgz-pJ8-org_json_TestJ8/src/org/agetac/common/model/impl/Source.java

```diff
@@ -0,0 +1,67 @@
+package org.agetac.common.model.impl;
+
+import org.agetac.common.exception.InvalidJSONException;
+import org.agetac.common.model.impl.Cible.CibleType;
+import org.agetac.common.model.impl.Source.SourceType;
+import org.agetac.common.model.sign.AbstractModel;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Source extends AbstractModel {
+	
+	public enum SourceType {
+		WATER,
+		FIRE,
+		CHEM
+	}
+	
+	private SourceType type;
+	
+	public Source() {
+		super(null, null, new Position(0,0));
+		this.type = SourceType.WATER;
+	}
+	
+	public Source(String uid, Position position, SourceType type) {
+		super(uid, null, position);
+		this.type = type;
+	}
+
+	public Source(JSONObject json) throws InvalidJSONException {
+		super(json);
+		
+		try {
+			this.type = SourceType.valueOf(json.getString("type"));
+		} catch (JSONException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)